### PR TITLE
read_topic API

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

The only query patterns I can see in our use-cases are to read a topic to completion, after a given timestamp. So, I made a method in the `ApiWrapper` that does that and handles pagination for the caller.

### CODE REVIEW

File: xmtp_mls/src/api_client_wrapper.rs [hunk 1]
```diff
@@ -37,9 +40,56 @@ where
...
+
+            for envelope in &result.envelopes {
+                out.push(envelope.clone());
+            }
+
+            if result.envelopes.len() < page_size as usize || result.paging_info.is_none() {
+                break;
+            }
+
+            cursor = match result.paging_info.unwrap().cursor {
```
[the remark this answers is not in the document]
Comment (neekolas) [reply xmtp_mls/src/api_client_wrapper.rs:74]: There is a chance that there is a null paging info for the last page. But that gets checked above and we break if it's `None`. I like that little tweak. Will make the change.

**Comment by richardhuaaa** [comment]
Nooby question - is there a max page size our API supports? If our client is always reading to completion, what is the purpose of paging under the hood?

**Comment by neekolas** [comment]
@richardhuaaa 100 item max page size

File: xmtp_mls/src/api_client_wrapper.rs [hunk 2]
```diff
@@ -34,6 +40,51 @@ where
...
+                        cursor,
+                        limit: page_size,
+                        direction: SortDirection::Ascending as i32,
+                    }),
+                })
+                .await?;
+
+            for envelope in &result.envelopes {
+                out.push(envelope.clone());
+            }
```
Comment (richardhuaaa) [inline xmtp_mls/src/api_client_wrapper.rs:68]: Small nit: I think you can do `out.append(result.envelopes)` if you do `let mut result` above and store the count beforehand, that might save a number of clones

File: xmtp_mls/src/api_client_wrapper.rs [hunk 3]
```diff
@@ -34,6 +40,51 @@ where
...
+
+            if result.envelopes.len() < page_size as usize || result.paging_info.is_none() {
+                break;
+            }
+
+            cursor = result
+                .paging_info
+                .expect("Empty paging info")
+                .cursor
+                .map(|wrapper| wrapper);
```
Comment (richardhuaaa) [inline xmtp_mls/src/api_client_wrapper.rs:78]: Don't forget this lint!

**Review by richardhuaaa — APPROVED** [review]
👍

File: xmtp_mls/src/api_client_wrapper.rs (hunk 2 above)
Comment (neekolas) [reply xmtp_mls/src/api_client_wrapper.rs]: That's pretty slick. Done

**Pull request merged**